docs/docsite/rst/network/user_guide/platform_eos.rst

```diff
@@ -0,0 +1,133 @@
++---------------------------+-----------------------------------------------+-----------------------------------------+
+
+
+Using CLI in Ansible 2.5
+================================================================================
+
+Example CLI ``group_vars/eos.yml``
+----------------------------------
+
+.. code-block:: yaml
+
+   ansible_connection: network_cli
+   ansible_network_os: eos
+   ansible_user: myuser
+   ansible_ssh_pass: !vault...
+   ansible_become: yes
+   ansible_become_method: enable
+   ansible_become_pass: !vault...
+   ansible_ssh_common_args: '-o ProxyCommand="ssh -W %h:%p -q bastion01"'
+
+
+- If you are using SSH keys (including an ssh-agent) you can remove the ``ansible_ssh_pass`` configuration.
+- If you are accessing your host directly (not through a bastion/jump host) you can remove the ``ansible_ssh_common_args`` configuration.
+- If you are accessing your host through a bastion/jump host, you cannot include your SSH password in the ``ProxyCommand`` directive. To prevent secrets from leaking out (for example in ``ps`` output), SSH does not support providing passwords via environment variables.
+
+Example CLI Task
+----------------
+
+.. code-block:: yaml
+
+   - name: Backup current switch config (eos)
+     eos_config:
+       backup: yes
+     register: backup_eos_location
+     when: ansible_network_os == 'eos'
+
+
+
+Using eAPI in Ansible 2.5
+================================================================================
+
+Enabling eAPI
+-------------
+
+Before you can use eAPI to connect to a switch, you must enable eAPI. To enable eAPI on a new switch via Ansible, use the ``eos_eapi`` module via the CLI connection. Set up group_vars/eos.yml just like in the CLI example above, then run a playbook task like this:
+
+.. code-block:: yaml
+
+   - name: Enable eAPI
+      eos_eapi:
+          enable_http: yes
+          enable_https: yes
+      become: true
+      become_method: enable
+      when: ansible_network_os == 'eos'
+
+You can find more options for enabling HTTP/HTTPS and local http in the :ref:`eos_eapi <eos_eapi>` module documentation.
+
+Once eAPI is enabled, change your ``group_vars/eos.yml`` to use the eAPI connection.
+
+Example eAPI ``group_vars/eos.yml``
+-----------------------------------
+
+.. code-block:: yaml
+
+   ansible_connection: local
+   ansible_network_os: eos
+   ansible_user: myuser
+   ansible_pass: !vault... 
+   eapi:
+     host: "{{ inventory_hostname }}"
+     transport: eapi
+     authorize: yes
+     auth_pass: !vault...
+   proxy_env:
+     http_proxy: http://proxy.example.com:8080
+
+- If you are accessing your host directly (not through a web proxy) you can remove the ``proxy_env`` configuration.
+- If you are accessing your host through a web proxy using ``https``, change ``http_proxy`` to ``https_proxy``.
+
+
+Example eAPI Task
+-----------------
+
+.. code-block:: yaml
```

Comment:
This isn't necessary, but it might be a nice quality of life improvement if this was a full playbook so that the user can just copy paste it directly to a file. 

```
---
- hosts: eos
  tasks:
  - name: Backup current switch config (eos)
    eos_config:
      backup: yes
      provider: "{{ eapi }}"
    register: backup_eos_location
    environment: "{{ proxy_env }}"
    when: ansible_network_os == 'eos'
```

Reply:
Thanks for the feedback. Quality-of-life and ease-of-use improvements are definitely on the roadmap but not a top priority for this PR.